docs/extend/server-schemas.md

```diff
@@ -0,0 +1,64 @@
+# Server component schemas
+
+Starting with Zowe v2.0, each Component in Zowe must contain a [json schema](https://json-schema.org/) describing the configuration parameters that are valid for its component section in Zowe's server configuration. If a component does not have anything that can be configured, this file can just be boilerplate specifying that it fully inherits generic Component parameters and nothing more.
+
+The server infrastructure will utilize each components' schema files to validate a Zowe instance configuration every startup, so this requirement is enforced by code.
+
+## Requirements
+
+* Server component json schema files must follow the json schema [spec 2019-09](https://json-schema.org/draft/2019-09/schema).
+* Each component must state where its base schema file is located by the manifest parameter "schemas.config"
+* The schema file must use and/or extend the Zowe Component base schema by use of the "allOf" attribute.
+* The schema must have an `$id` property which is a URI that has a domain related to the entity that developed the Component.
+* The file should be tagged on z/OS but elsewhere must at least be encoded as ASCII-subset of UTF-8
+
+## Additional information
+
+* The schema file can reference other schema files within the component if compartmentalization of definitions are desired
+
+
+## Example
+
+Below is an example manifest and schema for a Component named "component1". The manifest file specifies the location of the schema file, and the schema file specifies the configuration parameters that are valid for this Component.
+
+### Example manifest
+
+```
+name: component1
+id: com.abcdef.component1
+title: component1
+description: An example component with a simple configuration.
+license: ABCDEF Company License
+schemas:
+  - config: schema.json
```

Comment:
Is the `-` needed? Just try to understand the correct format. With dash, it means:

```json
{
  "schemas": [
    {
      "config": "schema.json"
    }
  ]
}
```

Reply:
oh, i messed this up. yes i'll remove the dash.